File: app/lib/font_weight_preferences.dart
import 'package:shared_preferences/shared_preferences.dart';

import 'typography.dart';

const _keyFontWeight = 'ultrasend_font_weight';

Future<FontWeightLevel> loadFontWeightLevel() async {
  final prefs = await SharedPreferences.getInstance();
  return decodeFontWeightLevel(prefs.getString(_keyFontWeight));
}

Future<void> persistFontWeightLevel(FontWeightLevel level) async {
  final prefs = await SharedPreferences.getInstance();
  await prefs.setString(_keyFontWeight, encodeFontWeightLevel(level));
}

File: web/src/lib/fontWeightPreferences.ts
/** Font weight preferences. Align key with Flutter `font_weight_preferences.dart`. */

import {
  decodeFontWeightLevel,
  encodeFontWeightLevel,
  type FontWeightLevel,
} from '@/lib/typography';

export const KEY_FONT_WEIGHT = 'ultrasend_font_weight';
export const FONT_WEIGHT_CHANGED_EVENT = 'ultrasend:font-weight-changed';

export function loadFontWeightLevel(): FontWeightLevel {
  if (typeof window === 'undefined') {
    return 'normal';
  }
  return decodeFontWeightLevel(localStorage.getItem(KEY_FONT_WEIGHT));
}

export function persistFontWeightLevel(level: FontWeightLevel): void {
  localStorage.setItem(KEY_FONT_WEIGHT, encodeFontWeightLevel(level));
  window.dispatchEvent(new CustomEvent(FONT_WEIGHT_CHANGED_EVENT));
}
